nexus/tests/integration_tests/oximeter.rs

```diff
@@ -146,12 +146,52 @@ async fn test_oximeter_reregistration() {
     //
     // At this point, Oximeter will still be collecting, so we poll for data and verify that
     // there's no new data in ClickHouse.
+    //
+    // Specifically, we grab a timestamp before dropping the producer, and assert that any data in
+    // the timeseries is from before that timestamp.
+    let time_producer_dropped = chrono::Utc::now();
```

Comment:
is it worth noting that a very narrow race still exists insofar as data might be recorded between this line and the actual `drop`? ... if I understand that correctly that is...